lib/puppet/provider/nameservice.rb

```diff
@@ -261,6 +269,10 @@ def groups
     groups.join(",")
   end
 
+  def groups
+    @groups ||= fetch_groups
```

Comment:
I'd rather not add caching to the provider at this late stage. The original ticket is about groups from AD containing spaces, and I'm thinking the least risky thing is to only validate that.